src/solver/optimisation/opt_construction_variables_optimisees_lineaire.cpp

```diff
@@ -104,11 +105,22 @@ void OPT_ConstruireLaListeDesVariablesOptimiseesDuProblemeLineaire(PROBLEME_HEBD
                   = VARIABLE_BORNEE_DES_DEUX_COTES;
                 NombreDeVariables++;
                 // 3. Level
-                CorrespondanceVarNativesVarOptim->ShortTermStorage.LevelVariable[globalIndex]
-                  = NombreDeVariables;
-                ProblemeAResoudre->TypeDeVariable[NombreDeVariables]
-                  = VARIABLE_BORNEE_DES_DEUX_COTES;
-                NombreDeVariables++;
+                if (pdt == 0 || (pdt % storage.storagecycle != 0))
```

Comment:
> Moreover, I can't see the point of the else block : if we're dealing with a level at the beginning / end of a cycle, it should show up in some constraint RHS, but we don't need to define a optimization variable, right ?

If the level is fixed, we could indeed set it as the RHS. But it would add even more complexity.